src/cargo/util/semver_ext.rs

```diff
@@ -111,16 +121,13 @@ impl OptVersionReq {
         }
     }
 
-    /// Since Semver does not support prerelease versions,
-    /// the simplest implementation is taken here without comparing the prerelease section.
-    /// The logic here is temporary, we'll have to consider more boundary conditions later,
-    /// and we're not sure if this part of the functionality should be implemented in semver or cargo.
+    /// Allows to match pre-release in SemVer-Compatible way.
+    /// See [`semver_eval_ext`] for matches_prerelease semantics.
     pub fn matches_prerelease(&self, version: &Version) -> bool {
-        if version.is_prerelease() {
-            let mut version = version.clone();
-            version.pre = semver::Prerelease::EMPTY;
-            return self.matches(&version);
+        if let OptVersionReq::Req(req) = self {
+            return req.matches_prerelease(version);
         }
+
         self.matches(version)
```

Comment:
nit: these two are mutually exclusive in their business logic (and its not just a guard statement).  For myself, I find using an `else` clarifies this relationship over an early return

Reply:
Done. Thank you.